test/unit/master/test_cluster_master.py

```diff
@@ -213,6 +213,38 @@ def test_updating_slave_to_nonexistent_state_should_raise_bad_request_error(self
         with self.assertRaises(BadRequestError):
             master.handle_slave_state_update(slave, 'NONEXISTENT_STATE')
 
+    def test_async_receive_heartbeat_from_slave_calls_set_heartbeat_for_the_slave(self):
+        master = ClusterMaster()
+        slave_url = "url"
+        slave_id = master.connect_slave(slave_url, 1)
+        slave = Mock()
+        master.get_slave = MagicMock(return_value=slave)
+        master._async_receive_heartbeat_from_slave(slave_id['slave_id'])
```

Comment:
Not a fan of calling a private method in a test. It's an indication that you're testing an implementation detail, which in turn is an indication that this test might be brittle. Brittle tests are a huge pet peeve of mine, and something that this team has suffered heavily from in the past.

The easiest way to fix this test is to call `master.receive_heartbeat_from_slave` directly, patch the slave constructor, and assert that when you call `master.receive_heartbeat_from_slave` the return value of the slave constructor (i.e. the slave) gets `set_heartbeat` called. 

An improvement upon that test is instead of calling `master.receive_heartbeat_from_slave`, send a POST to the heartbeat endpoint and assert that slave.set_heartbeat is called tornado's [AsyncHTTPTestCase](http://www.tornadoweb.org/en/stable/testing.html#tornado.testing.AsyncHTTPTestCase) will be helpful here. Look at `get_app()` and `fetch`

See the internal box code that I slack'd to you